spec/controllers/equipment_models_controller_spec.rb

```diff
@@ -1,421 +1,273 @@
-    end
-  end
-  context 'with show deleted' do
-    let!(:mod_other_cat_active) { FactoryGirl.create(:equipment_model) }
-    let!(:mod_other_cat_inactive) do
-      FactoryGirl.create(:equipment_model,
-                         deleted_at: Time.zone.today)
-    end
-    context 'with @category set' do
-      it 'should populate an array of category-type equipment models' do
-        mod_same_cat_inactive =
-          FactoryGirl.create(:equipment_model, category: model.category,
-                                               deleted_at: Time.zone.today)
-        get :index, category_id: model.category, show_deleted: true
-        expect(assigns(:equipment_models).include?(model)).to be_truthy
-        expect(assigns(:equipment_models).include?(mod_other_cat_active))
-          .not_to be_truthy
-        expect(assigns(:equipment_models).include?(mod_same_cat_inactive))
-          .to be_truthy
-        expect(assigns(:equipment_models).include?(mod_other_cat_inactive))
-          .not_to be_truthy
-        expect(assigns(:equipment_models).size).to eq(2)
-      end
-    end
-    context 'without @category set' do
-      it 'should populate an array of all equipment models' do
-        get :index, show_deleted: true
-        expect(assigns(:equipment_models).include?(model)).to be_truthy
-        expect(assigns(:equipment_models).include?(mod_other_cat_active))
-          .to be_truthy
-        expect(assigns(:equipment_models).include?(mod_other_cat_inactive))
-          .to be_truthy
-        expect(assigns(:equipment_models).size).to eq(3)
-      end
-    end
-  end
-end
-
 describe EquipmentModelsController, type: :controller do
-  before(:each) { mock_app_config(requests_affect_availability: false) }
-  let!(:model) { FactoryGirl.create(:equipment_model) }
+  include EquipmentModelHelper
 
+  before(:each) { mock_app_config(requests_affect_availability: false) }
   it_behaves_like 'calendarable', EquipmentModel
 
+  USER_ROLES = [:admin, :user].freeze
+
   describe 'GET index' do
-    context 'with admin user' do
+    shared_examples_for 'GET index success' do |user_role|
       before do
-        sign_in FactoryGirl.create(:admin)
-        get :index
+        mock_user_sign_in(mock_user(user_role))
       end
-      it_behaves_like 'GET index success'
-    end
-    context 'with non-admin user' do
-      before { sign_in FactoryGirl.create(:user) }
-      describe 'should redirect to root' do
+
+      describe 'basic function' do
         before { get :index }
-        it_behaves_like 'GET index success'
+        it_behaves_like 'successful request', :index
+      end
+
+      it 'defaults to all active equipment models' do
+        # UNSAFE, but a stand in for a relation
+        models = spy('Array')
+        # get around the eager loading
+        allow(models).to receive(:includes).and_return(models)
+        allow(EquipmentModel).to receive(:where).and_return([])
+        expect(EquipmentModel).to receive(:all).and_return(models)
```

Comment:
Can we be consistent with whether we're setting expectations prior to the action or following it? My preference is following since it more clearly represents the four-phase test model (e.g. setup --> exercise --> verify), but either way within a specific test I feel like we should be consistent.